src/pip/_internal/vcs/__init__.py

```diff
@@ -186,6 +203,55 @@ class VersionControl(object):
     unset_environ = ()  # type: Tuple[str, ...]
     default_arg_rev = None  # type: Optional[str]
 
+    @classmethod
+    def should_add_vcs_url_prefix(cls, remote_url):
+        """
+        Return whether the vcs prefix (e.g. "git+") should be added to a
+        repository's remote url when used in a requirement.
+        """
+        return not remote_url.lower().startswith('{}:'.format(cls.name))
+
+    @classmethod
+    def get_subdirectory(cls, repo_dir):
+        """
+        Return the path to setup.py, relative to the repo root.
```

Comment:
```suggestion
        Return the path to setup.py/pyproject.toml, relative to the repo root.
```

Reply:
Thanks for reviewing, @xavfernandez! Good observation here. The method's current implementation, though, only looks for `setup.py`: https://github.com/pypa/pip/blob/5f3c56e1889ccafe49633e7bc7871b3b34ce2a8a/src/pip/_internal/vcs/git.py#L289-L314

Maybe this means you found a bug, or at least an opportunity for a future enhancement?

Reply:
Indeed :+1: